### ISSUE TITLE
Some frameworks are overwriting the mongoDB world table: ffead-cpp, kelp, sabina

### ISSUE THREAD

**Issue by knewmanTE** [TechEmpower/FrameworkBenchmarks#2235]
Before running certain tests test, I am seeing results like this in my local mongoDB database (freshly dropped and rebuilt):

```
{ "_id" : 1, "id" : 1, "randomNumber" : 7229 }
{ "_id" : 2, "id" : 2, "randomNumber" : 5321 }
{ "_id" : 3, "id" : 3, "randomNumber" : 9144 }
{ "_id" : 4, "id" : 4, "randomNumber" : 9358 }
{ "_id" : 5, "id" : 5, "randomNumber" : 4414 }
{ "_id" : 6, "id" : 6, "randomNumber" : 5760 }
{ "_id" : 7, "id" : 7, "randomNumber" : 6196 }
{ "_id" : 8, "id" : 8, "randomNumber" : 5389 }
{ "_id" : 9, "id" : 9, "randomNumber" : 5945 }
{ "_id" : 10, "id" : 10, "randomNumber" : 6208 }
{ "_id" : 11, "id" : 11, "randomNumber" : 7997 }
{ "_id" : 12, "id" : 12, "randomNumber" : 6508 }
{ "_id" : 13, "id" : 13, "randomNumber" : 7238 }
{ "_id" : 14, "id" : 14, "randomNumber" : 834 }
{ "_id" : 15, "id" : 15, "randomNumber" : 153 }
{ "_id" : 16, "id" : 16, "randomNumber" : 9355 }
{ "_id" : 17, "id" : 17, "randomNumber" : 6444 }
{ "_id" : 18, "id" : 18, "randomNumber" : 5860 }
{ "_id" : 19, "id" : 19, "randomNumber" : 974 }
{ "_id" : 20, "id" : 20, "randomNumber" : 596 }
```

After running the tests, I am seeing results like this:
 ```
{ "_id" : 1, "id" : 1, "randomNumber" : 7776 }
{ "_id" : 2, "id" : 2, "randomNumber" : 7637 }
{ "_id" : 3, "randomNumber" : 3636 }
{ "_id" : 4, "randomNumber" : 4423 }
{ "_id" : 5, "id" : 5, "randomNumber" : 9595 }
{ "_id" : 6, "randomNumber" : 5993 }
{ "_id" : 7, "id" : 7, "randomNumber" : 3051 }
{ "_id" : 8, "id" : 8, "randomNumber" : 5884 }
{ "_id" : 9, "randomNumber" : 3341 }
{ "_id" : 10, "randomNumber" : 8440 }
{ "_id" : 11, "randomNumber" : 1609 }
{ "_id" : 12, "randomNumber" : 76 }
{ "_id" : 13, "id" : 13, "randomNumber" : 4010 }
{ "_id" : 14, "randomNumber" : 2314 }
{ "_id" : 15, "randomNumber" : 9219 }
{ "_id" : 16, "randomNumber" : 1355 }
{ "_id" : 17, "randomNumber" : 282 }
{ "_id" : 18, "randomNumber" : 4697 }
{ "_id" : 19, "id" : 19, "randomNumber" : 1417 }
{ "_id" : 20, "randomNumber" : 3134 } 
```

They appear to be using the <b>_id</b> field instead of the <b>id</b> and then overwriting the previous entry. I believe this may be causing some other MongoDB tests to subsequently break, as they might expect an "id" field and receive an item that doesn't have it.

So far, the problematic tests have been:

- ffead-cpp (editing the <i>World</i> table)
- kelp and kelp-mongodb (editing the <i>World</i> table)
- sabina (edited a single entry in the <i>world</i> table, though I haven't been able to get it to consistently reproduce)

**Comment by msmith-techempower** [comment]
:sigh:

@knewmanTE Can you investigate the equivalent of `GRANT SELECT, UPDATE ON...` in mongodb and then update our script to include that? Basically, the schema should be unalterable.

**Comment by knewmanTE** [comment]
@msmith-techempower I'm not sure if that is possible, given that MongoDB doesn't have a set schema for a database, but I'll keep looking around to see if I find anything along the lines of what we want.

**Comment by knewmanTE** [comment]
ping @naturalist @sumeetchhetri @jaguililla

**Comment by jaguililla** [comment]
Hello, Sabina framework is no longer maintained and I made a PR to delete it. Anyway, it was rejected for the Round 13.

About the problem, in the DTOs of my tests, the `id` field is mapped to `_id` to avoid duplicated fields (model `id` <-> mongodb `_id`).

This way in the 'update test' the row is overwritten with my test's model (losing the `id` field).

This can cause problems (duplicate IDs) sorry I haven't foreseen this before.

Anyway, It would be nice to isolate each test, loading a fresh DB for each for the sake of reliability.

Could you please remove the framework? as I do not have time now to fix it and I'm more interested in Round 14 for a new framework (Hexagon in Kotlin).

Thanks for your great work!

**Comment by knewmanTE** [comment]
@jaguililla sorry about that. I was looking back at the PR and it looks like there was some branching confusion. We'll remove sabina for the Round 13 Final run!

**Comment by jaguililla** [comment]
That's ok... could you also apply the other PR I did? To include the Hexagon framework ;)

**Comment by knewmanTE** [comment]
@jaguililla I believe Hexagon is already merged in! (see [here](https://github.com/TechEmpower/FrameworkBenchmarks/tree/master/frameworks/Kotlin/hexagon)).

We merged it into round-14 in early August, but ended up folding most of our round-14 changes into master (round 13) because there were some unexpected delays in getting round 13 results out to the public.

**Comment by jaguililla** [comment]
Hey... that's great! thank you very much!!!

**Comment by knewmanTE** [comment]
Hey @jaguililla, I've noticed that Hexagon has been failing to start up lately because of an issue with the Gradle Daemon (notably the plugin "org.hidetake.ssh"). Any chance you could take a look and see if it's a quick fix? This is the output I see when running a hexagon test:
```
Setup hexagon: Starting a new Gradle Daemon for this build (subsequent builds will be faster).
Setup hexagon: 
Setup hexagon: FAILURE: Build failed with an exception.
Setup hexagon: 
Setup hexagon: * Where:
Setup hexagon: Script 'https://raw.githubusercontent.com/jaguililla/hexagon/master/gradle/hexagon_service.gradle' line: 4
Setup hexagon: 
Setup hexagon: * What went wrong:
Setup hexagon: A problem occurred evaluating script.
Setup hexagon: > Plugin with id 'org.hidetake.ssh' not found.
Setup hexagon: 
Setup hexagon: * Try:
Setup hexagon: Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
Setup hexagon: 
Setup hexagon: BUILD FAILED
```

**Comment by jaguililla** [comment]
My fault... let me fix it tomorrow

**Comment by jaguililla** [comment]
Thanks for noticing me

**Comment by jaguililla** [comment]
I upgraded the framework and scripts to the latest versions (TechEmpower/FrameworkBenchmarks#2280)  @knewmanTE

**Comment by knewmanTE** [comment]
No longer seeing this problem with ffead-cpp. Sabina resolved with #2268 and kelp resolved with #2276.

Closing this issue.
